### PR TITLE
storage/engine: add cf interface.

### PR DESCRIPTION
**Pull request by disksing**
@siddontang @BusyJay @ngaut @zhangjinpeng1987

### CODE REVIEW

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -376,7 +376,7 @@ mod tests {
 
     #[test]
     fn test_mvcc_txn_commit_err() {
-        let engine = engine::new_engine(Dsn::RocksDBPath(TEMP_DIR)).unwrap();
+        let engine = engine::new_engine(Dsn::RocksDBPath(TEMP_DIR), DEFAULT_ENGINE_CFS).unwrap();
```
Comment (ngaut) [inline src/storage/mvcc/txn.rs]: s/DEFAULT_ENGINE_CFS/DEFAULT_CFS/

File: src/storage/engine/mod.rs [hunk 2]
```diff
@@ -151,10 +162,10 @@ pub enum Dsn<'a> {
 }
 
 // Now we only support RocksDB.
-pub fn new_engine(dsn: Dsn) -> Result<Box<Engine>> {
+pub fn new_engine(dsn: Dsn, cfs: &'static [CfName]) -> Result<Box<Engine>> {
```
Comment (siddontang) [inline src/storage/engine/mod.rs]: need static here?

Comment (disksing) [inline src/storage/engine/mod.rs]: Not necessary indeed.

**Comment by ngaut** [comment]
PTAL @BusyJay

File: src/storage/engine/mod.rs [hunk 3]
```diff
@@ -31,22 +31,33 @@ const SEEK_BOUND: usize = 30;
 #[derive(Debug)]
 pub enum Modify {
     Delete(Key),
```
Comment (BusyJay) [inline src/storage/engine/mod.rs]: Can we just keep the cf version?

File: src/storage/engine/rocksdb.rs [hunk 4]
```diff
@@ -39,15 +40,37 @@ impl EngineRocksdb {
...
+            temp_dir: temp_dir,
+        })
+    }
+
+    fn open_or_create_db(path: &str, cfs: &[CfName]) -> Result<DB> {
+        let mut opts = Options::new();
+        opts.create_if_missing(false);
+        match DB::open_cf(&opts, path, cfs) {
+            Ok(db) => return Ok(db),
+            Err(e) => warn!("open rocksdb fail: {}", e),
```
Comment (BusyJay) [inline src/storage/engine/rocksdb.rs]: Why fallback? I think we should return `Err` here.

Comment (disksing) [inline src/storage/engine/rocksdb.rs]: @BusyJay Rocksdb requires to open with all existing column family names. So we first try to open an old db (with all column families we want), then try to create one if step 1 fails.

File: src/storage/engine/mod.rs (hunk 3 above)
Comment (disksing) [inline src/storage/engine/mod.rs]: Good suggestion, I'll have a try.

Comment (disksing) [inline src/storage/engine/mod.rs]: @BusyJay Just reconsidered about it. Currently we use `Modify::Delete`/`Modify::Put` to construct a Modify in many places, keep the old version will make them more simple.

**Comment by disksing** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**